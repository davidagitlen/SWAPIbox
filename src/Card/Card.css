.Card {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  height: 300px;
  width: 250px;
  margin: 10px 25px 10px 25px; 
  padding: 10px;
  color: ghostwhite;
  background-color: #000A11;
  border-radius: 10px;
  box-shadow: 10px 10px 5px rgba(63, 195, 128, .2);
}

.Card-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-content: center;
  height: 100px;
  width: 100%;
  border: 1px solid ghostwhite;
}

.stats {
  display: flex;
  flex-direction: column;
}

/* p {
  margin: 10px 10px 10px 10px;
} */

button {
  height: 25px;
  width: 75px;
  border-radius: 10px;
}