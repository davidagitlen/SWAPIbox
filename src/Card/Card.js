import React from 'react';
import './Card.css'
import hilt from '../images/non-chunky-lightsaber-hilt.png'
import saber from '../images/non-chunky-lightsaber.png'

const Card = ({ name, homeworld, species, population, terrain, climate, residents, model, vehicleClass, numberOfPassengers, isFavorite, favoriteCard, type }) => {

  const handleClick = e => {
    const cardname = e.target.getAttribute('cardname');
    const cardtype = e.target.getAttribute('cardtype');
    const favorite = e.target.getAttribute('favorite');
    favoriteCard(cardname, cardtype, favorite);
  } 
  
  const saberSource = isFavorite === 'true' ? saber : hilt;
  return (
    <section className='Card'>
      <header className='Card-header'>
        <h2>{name}</h2>
        <label htmlFor='favorite'>Favorite
          <input type='image' id='hilt' name='favorite' cardname={name} cardtype={type} favorite={isFavorite} src={saberSource} onClick={handleClick}/>
        </label>
      </header>
      <div className='stats'>
        {homeworld && <p>Homeworld: {homeworld}</p>}
        {species && <p>Species: {species}</p>}
        {population && <p>Population: {parseInt(population).toLocaleString()}</p>}
        {terrain && <p>Terrain: {terrain}</p>}
        {climate && <p>Climate: {climate}</p>}
        {residents && <p>Residents: {residents}</p>}
        {model &&<p>Model: {model}</p>}
        {vehicleClass && <p>Class: {vehicleClass}</p>}
        {numberOfPassengers &&<p>Capacity: {numberOfPassengers} Passengers</p>}
      </div>
    </section>
  )
}

export default Card;