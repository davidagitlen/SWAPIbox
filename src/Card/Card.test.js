import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card'

describe('Card', () => {
  let wrapper, favoriteCardMock, mockData;

  beforeEach(() => {
    mockData = {
    name : 'Eric' ,
    homeworld : 'Earth', 
    species : 'Human', 
    population : '10',
    terrain : 'rocky',
    climate : 'temperate',
    residents : ['Eric', 'David', 'Chris'], 
    model : 'jalopy' ,
    vehicleClass : 'jalopy',
    numberOfPassengers : 4, 
    type : 'people'
    };
    favoriteCardMock = jest.fn();
    wrapper = shallow(<Card 
      data = {mockData}
      favoriteCard = {favoriteCardMock}
      favoritesArray = {[]}
      />)
  });

  it('should render without crashing', () => {
    expect(wrapper.length).toEqual(1)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();


  });

  it('should call favoriteCard when the favorite input is clicked', () => {
    wrapper.find('input').simulate('click');
    expect(favoriteCardMock).toHaveBeenCalled();
    expect(favoriteCardMock).toHaveBeenCalledWith(mockData);
  })

  // it('should call favoriteCard when the favorite input is clicked', () => {
  //   const wrapper = mount(<Card
  //     data={mockData}
  //     favoriteCard={jest.fn()}
  //     favoritesArray={[]}
  //   />)
  //   wrapper.find('input').simulate('click');
  //   expect(wrapper.props().favoriteCard).toHaveBeenCalled();
  //   expect(wrapper.props().favoriteCard).toHaveBeenCalledWith(mockData);
  // });

});