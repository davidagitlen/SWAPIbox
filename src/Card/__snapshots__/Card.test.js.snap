// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<section
  className="Card"
>
  <header
    className="Card-header"
  >
    <div
      className="Card-name"
    >
      <h2>
        Eric
      </h2>
    </div>
    <div
      className="favorite-button"
    >
      <input
        alt="active/inactive lightsaber showing when card is favorited"
        cardname="Eric"
        cardtype="people"
        id="hilt"
        name="favorite"
        onClick={[Function]}
        src="non-chunky-lightsaber-hilt.png"
        type="image"
      />
      <label
        htmlFor="favorite"
      >
        Favorite
      </label>
    </div>
  </header>
  <div
    className="stats"
  >
    <p>
      HOMEWORLD: 
      Earth
    </p>
    <p>
      SPECIES: 
      Human
    </p>
    <p>
      POPULATION: 
      10
    </p>
    <p>
      TERRAIN: 
      rocky
    </p>
    <p>
      CLIMATE: 
      temperate
    </p>
    <div>
      RESIDENTS :
    </div>
     
    <ul
      className="resident-list"
    >
      <ul
        className="resident-list"
      >
        <li
          key="Eric"
        >
          Eric
        </li>
        <li
          key="David"
        >
          David
        </li>
        <li
          key="Chris"
        >
          Chris
        </li>
      </ul>
    </ul>
    <p>
      MODEL: 
      jalopy
    </p>
    <p>
      CLASS: 
      jalopy
    </p>
    <p>
      CAPACITY: 
      4
       Passengers
    </p>
  </div>
</section>
`;
